src/storage/txn/latch.rs

```diff
@@ -29,6 +30,49 @@ impl Latch {
             waiting: VecDeque::new(),
         }
     }
+
+    pub fn front(&self, s: usize) -> Option<u64> {
+        for item in self.waiting.iter() {
+            if let Some((x, y)) = item {
+                if *x == s {
+                    return Some(*y);
+                }
+            }
+        }
+        None
+    }
+
+    pub fn pop_front(&mut self, s: usize) -> Option<(usize, u64)> {
+        for item in self.waiting.iter_mut() {
+            if let Some((v, _)) = item {
+                if *v == s {
+                    return item.take();
+                }
+            }
+        }
+        None
+    }
+
+    pub fn wait_for_wake(&mut self, s: usize, cid: u64) {
+        self.waiting.push_back(Some((s, cid)));
+    }
+
+    pub fn maybe_shrink(&mut self) {
+        // Pop item which is none to make queue not too long.
+        while !self.waiting.is_empty() {
+            let item = self.waiting.front().unwrap();
+            if item.is_none() {
+                self.waiting.pop_front();
+            } else {
+                break;
```

Comment:
too complexed? How about the performance with other solutions like make Latches to a Concurrency hash map?

Reply:
+1

Reply:
No. ConcurrencyHashmap could only support simple operation such as `get` and `put`, but here we need a queue could store request id which has the same key.

Reply:
If you use ConcurrencyHashMap, you still need write such code:
```
latches: ConcurrencyHashMap<Key, Mutex<VecDeque<u64>>;
```
Obviously, `ConcurrencyHashMap` would perform much worse than array.

Reply:
If you want a better performance than this PR, you maybe need a ConcurrencyHashMap and ConcurrencyQueue. I think it would be more complex to deal with write-request in different thread.

Reply:
How do you think it compares to `ConcurrentHashMap<ArcKey, Task>`? That's one more allocation but the benefits is simpler latch, less memory usage and low initialization cost.

Reply:
But how do you deal with tasks which has the same key?  Don't you need to store them in queue?

Reply:
Since we must store requests with the same key in the same queue,  queue is necessary.  I just merge bucket of hashtable and queue of request into one structure.

Reply:
Why do we need concurrent data structures? Scheduler is single threaded if I remember correctly.